### DISCUSSION TITLE
Explicitly indicate ALAC support?

### DISCUSSION THREAD

**Discussion by mdrxy** [AzuraCast/AzuraCast#7336] [Polls]
While it isn't mentioned anywhere on the web interface or in the docs, ALAC files are supported by AzuraCast when FLAC is chosen as the AutoDJ format (remote DJ still untested on my end). Icecast treats the stream as FLAC, using FLAC_version 1.0 on the status page.

Would it be naive to more explicitly indicate this? Depending on how people feel, I'd be happy to make a PR.

**Comment by gAlleb** [comment]
@mdrxy, hi!

> While it isn't mentioned anywhere on the web interface or in the docs, ALAC files are supported by AzuraCast when FLAC is chosen as the AutoDJ format (remote DJ still untested on my end). Icecast treats the stream as FLAC, using FLAC_version 1.0 on the status page.

Alac files are supported. I think you're confusing decoders with encoders. https://www.liquidsoap.info/doc-dev/encoding_formats.html

You can have ALAC, wav, m4a, you name it files in your playlist. They are decoded by Liquidsoap into its internal format and then encoded into the format that you choose in the `output`.  Unless you explicitly ask ffmpeg to copy audio (`%audio.copy`) - then it will copy data without decoding or encoding it. But you won't be able to use LS operators to process that audio data anymore.

To sum up: you can choose any format in `AutoDJ` settings of the mount point and it will play ALAC files.

<hr>

P.S. Due to decoding/encoding process of LS - it's best to use FLAC/ALAC files as input. This way you'll get a "better" sound quality when using such encoders as mp3, aac, ogg. E.g in my setup I tend too use only flac and alac files. Beware that if you use initially already compressed audio (mp3, aac, etc) and an `mp3` output - you will not get the initial compressed quality of your mp3 files because they will be decoded and then encoded again. So, playing from good FLAC/ALAC sources is favourable.
